src/lit-html.ts

```diff
@@ -613,14 +617,25 @@ export class TemplateInstance {
         }
       }
     }
-    if (this.template.svg) {
-      const svgElement = fragment.firstChild!;
-      fragment.removeChild(svgElement);
-      const nodes = svgElement.childNodes;
-      for (let i = 0; i < nodes.length; i++) {
-        fragment.appendChild(nodes.item(i));
-      }
-    }
     return fragment;
   }
 }
+
+/**
+ * Reparents nodes, starting from `startNode` (inclusive) to `endNode`
+ * (exclusive), into another container (could be the same container), before
+ * `beforeNode`. If `beforeNode` is null, it appends the nodes to the
+ * container.
+ */
+export const reparentNodes =
+    (startNode: Node | null,
+     endNode: Node | null,
+     container: Node,
+     beforeNode: Node | null): void => {
+      let node = startNode;
+      while (node !== endNode) {
+        const n = node!.nextSibling;
+        container.insertBefore(node!, beforeNode);
```

Comment:
If we don't yet have a case of using `beforeNode`, I think that `appendChild` is 10-20% faster than `insertBefore` in Chrome due to a couple of additional precondition checks.

Reply:
It'll be necessary for the `repeat` directive, so I can move the current part before `currentMarker`.